lib/miq_automation_engine/service_models/miq_ae_service_orchestration_stack.rb

```diff
@@ -30,5 +31,10 @@ def normalized_live_status
     rescue MiqException::MiqOrchestrationStackNotExistError => err
       ['not_exist', err.message]
     end
+
+    def self.refresh(manager_id, manager_ref)
+      OrchestrationStack.refresh_ems(manager_id, manager_ref)
+      true
+    end
```

Comment:
@Ladas What is the return from `OrchestrationStack.refresh_ems`?  Why are you overriding it?

Reply:
I think that just queues full/targeted refresh based on settings, so it should not have any return value

Reply:
Right, so why override the return value?

I would suggest just wrapping the results:

```ruby
def self.refresh(manager_id, manager_ref)
  wrap_results(OrchestrationStack.refresh_ems(manager_id, manager_ref))
end
```

@mkanoor thoughts?

Reply:
Honestly, I don't know, tell me what is correct. :-)

I've seens this ```return true``` in some places, as a safe way to avoid non mapped object returned, if we don't care about return value.

Reply:
It would be helpful if you could tell us what is returned from the method you are calling, then we could make an informed decision.  I think that's the piece that's missing here.

Reply:
https://github.com/Ladas/manageiq/blob/300deb4c9aa3f7d35401f4a61fee980921142490/app/models/orchestration_stack.rb#L147

The queue refresh is the last command

Reply:
Correct.

> what is returned from the method you are calling

It looks like `EmsRefresh.queue_refresh` returns either an array or IDs or `nil`.  (https://github.com/ManageIQ/manageiq/blob/master/app/models/ems_refresh.rb#L63) 

We do not have to convert that return value into a service model then just change the method to:
```ruby
def self.refresh(manager_id, manager_ref)
  OrchestrationStack.refresh_ems(manager_id, manager_ref)
end
```

Reply:
@Ladas @gmcculloug 
if we can wrap the results of OrchestrationStack.refresh_ems(manager_id, manager_ref) we can just pass that thru. If its nil it should be ok.

Reply:
ok, removing the ```true```